### DISCUSSION TITLE
Pulling only new / changed records

### DISCUSSION THREAD

**Discussion by henda79** [Azure/azure-mobile-apps#435] [Q&A]
I've got a table with 2 rows and each time I pull, I get those 2 rows even though nothing has changed. Is this supposed to happen ? as I thought after a successful pull, the updatedAt filter would change so the next pull would return 0 rows.

I've had a look at the overloads for `PullItemsAsync` but can't find anything for that other than keep a track of this locally.

**Comment by adrianhall** [comment]
When you run PullItemsAsync, the internal table should be updated.  It is on my copy and there are tests on PullItemsAsync to check this, so I'm curious as to why it isn't happening on yours.  However, it will be updated to the updatedAt for the last record pulled. 

Maybe I should add a couple of tests to check incremental pulls more fully as an integration test.

**Reply by henda79** [reply]
Do you have any pointers on how I can dig into this a bit deeper. 

See this table for exmple

<table style="width: 459px;">
<tbody>
<tr>
<td style="width: 102.922px;">Id</td>
<td style="width: 133.516px;">UpdatedAt</td>
<td style="width: 56.4375px;">Version</td>
<td style="width: 35.25px;">Deleted</td>
<td style="width: 133.516px;">CreatedAt</td>
</tr>
<tr>
<td style="width: 102.922px;">b9bf0a73-d6a4-4527-b155-2245bf495ead</td>
<td style="width: 133.516px;">2022-07-18 18:29:32.4900000 +00:00</td>
<td style="width: 56.4375px;">&lt;Binary data&gt;</td>
<td style="width: 35.25px;">False</td>
<td style="width: 133.516px;">2022-07-18 18:29:32.4900000 +00:00</td>
</tr>
<tr>
<td style="width: 102.922px;">7748270c-be1d-41c4-bf9c-792fc7a17907</td>
<td style="width: 133.516px;">2022-07-18 18:29:04.0633333 +00:00</td>
<td style="width: 56.4375px;">&lt;Binary data&gt;</td>
<td style="width: 35.25px;">False</td>
<td style="width: 133.516px;">2022-07-18 18:29:04.0633333 +00:00</td>
</tr>
<tr>
<td style="width: 102.922px;">ae1f0742-a5b8-4106-bdd0-ac3a16fef2f9</td>
<td style="width: 133.516px;">2022-07-18 18:29:50.5166667 +00:00</td>
<td style="width: 56.4375px;">&lt;Binary data&gt;</td>
<td style="width: 35.25px;">False</td>
<td style="width: 133.516px;">2022-07-18 18:29:50.5166667 +00:00</td>
</tr>
<tr>
<td style="width: 102.922px;">51b0cfcc-2989-4534-84c8-bca2c9387196</td>
<td style="width: 133.516px;">2022-07-18 18:30:10.9833333 +00:00</td>
<td style="width: 56.4375px;">&lt;Binary data&gt;</td>
<td style="width: 35.25px;">False</td>
<td style="width: 133.516px;">2022-07-18 18:30:10.9833333 +00:00</td>
</tr>
</tbody>
</table>

On first pull I get 4 rows... perfect
on 2nd and every subsequent pull I get 1 row (**last row in table**).

Here is the query which is sent to the server

```
Query: ?$filter=(updatedAt%20gt%20cast(2022-07-18T18:30:10.000Z,Edm.DateTimeOffset))&__includedeleted=true
```

Notice the UpdatedAt time for the last row and the query. It seems the query truncates the time and therefore returns the row.

The data type in the server MSSQL Db is datetimeoffset(7), which seems to be the default for DateTimeOffset in c#

I suppose one workaround would be change the type in the server Db to datetimeoffset(0) but doing this on a table with data causes it to round up, so issue will still persist, however this is fine for empty.

The best solution would be to send the filter query with the correct precision.

What do you think Adrian ?

**Reply by adrianhall** [reply]
Totally agreed - I need to look into why the query doesn't have the right precision.  I'm moving this to an issue for investigation.

**Reply by henda79** [reply]
If I get time, I'll have a look further, but for now its not causing major issues as we are not in production.
